api/devicerepository.proto

```diff
@@ -303,11 +307,51 @@ message ListEndDeviceModelsResponse {
   repeated EndDeviceModel models = 1;
 }
 
-message MessagePayloadFormatter {
+message EncodedMessagePayload {
+  uint32 f_port = 1 [(validate.rules).uint32.lte = 255];
+  bytes frm_payload = 2 [(gogoproto.customname) = "FRMPayload"];
```

Comment:
Should we set a maximum length validation rule here?

Reply:
Leaving unset, since there are no validations rules in the `MACPayload` message either https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.12/api/lorawan.proto#L138